markdown.go

```diff
@@ -452,8 +453,9 @@ func secondPass(p *parser, input []byte) []byte {
 	if p.flags&EXTENSION_FOOTNOTES != 0 && len(p.notes) > 0 {
 		p.r.Footnotes(&output, func() bool {
 			flags := LIST_ITEM_BEGINNING_OF_LIST
-			for _, ref := range p.notes {
+			for i := 0; i < len(p.notes); i += 1 {
 				var buf bytes.Buffer
+				ref := p.notes[i]
```

Comment:
Style suggestion, I would move `ref := p.notes[i]` to the the very first thing after `for i := 0; i < len(p.notes); i += 1 {`, since you're effectively replacing the `ref` from `for _, ref := range p.notes {`. I think it's slightly cleaner that way, but I don't feel very strongly about it if you disagree.